packages/blitz-auth/src/server/passport-adapter.ts

```diff
@@ -108,15 +109,18 @@ export function passportAuth(config: BlitzPassportConfig): ApiHandler {
       "No Passport strategies found! Please add at least one strategy.",
     )
 
-    const blitzStrategy = configObject.strategies.find(
-      ({strategy}) => strategy.name === req.query?.auth?.[0] ?? "",
+    // find the requested strategy based on the manual specified name
+    // or if name is not specified, based on the strategy name
+    const blitzStrategy = configObject.strategies.find(({name, strategy}) =>
+      name ? name == req.query?.auth?.[0] : strategy.name === req.query?.auth?.[0] ?? "",
     )
     assert(blitzStrategy, `A passport strategy was not found for: ${req.query.auth[0]}`)
 
-    const {strategy, authenticateOptions} = blitzStrategy
+    const {name, strategy, authenticateOptions} = blitzStrategy
 
-    passport.use(strategy)
-    const strategyName = strategy.name as string
+    const strategyName = (name || strategy.name) as string
+
```

Comment:
Let's add this assertion to get rid of `as string` when assigning `strategyName`:

```suggestion
    assert(typeof strategy.name  !== "undefined", `A passport strategy name was not found for: ${req.query.auth[0]}`)
    
    const strategyName = name || strategy.name
```